### ISSUE TITLE
Using filter(PShader) will bug blending mode

### ISSUE THREAD

**Issue by iGoodie** [processing/processing#5616]
## Description
I'm trying to postprocess my screenbuffer with a basic vignette fragment shader.
I load the shader, then use `filter()`, the very first frame is perfect, but the other one is just blended wrong ignoring the transparency values.

## Expected Behavior
`image()` callbacks will blend properly after `filter(PShader)` callbacks.
Very first frame drawn;
![proper_frame](https://cdn.pbrd.co/images/HyGx6HA.png)

## Current Behavior
`image()` renders images ignoring the transparency values
Frames after the first frame drawn;
![improper_frame](https://cdn.pbrd.co/images/HyGymb2.png)

## Steps to Reproduce

1. Load following `vignette.fsh` shader;
```glsl
#ifdef GL_ES
precision mediump float;
precision mediump int;
#endif

#define PROCESSING_TEXTURE_SHADER

uniform sampler2D texture;
uniform vec2 resolution;
 
varying vec4 vertColor;
varying vec4 vertTexCoord;

void main() {
	vec4 texColor = texture2D(texture, vertTexCoord.xy);
	
	vec2 position = (gl_FragCoord.xy / resolution.xy) - vec2(0.5);
	
	float len = length(position);
	
	gl_FragColor = vec4( texColor.rgb*(1.0-len), 1.0 );
}
```
2. Execute given `draw()` method body;
```java
public void render() {
	background(255);
	image(someTransImg);
	filter(vignetteShader);
}
```
3. See it will draw transparency right only for the first frame. (use `frameRate(1)` @ `setup()` to see easier)

## Your Environment

 

* Processing version: 3.4 (Stable)
* Operating System and OS version: Windows 7 Home Premium

## Possible Causes / Solutions
I believe `filter(PShader)` somehow changes the blending mode into MIX mode.
Also calling `blendMode(BLEND)` right after `filter(PShader)` solves it. But that shouldn't be an intended behavior

**Comment by Sardtok** [comment]
Could you test with Processing 3.3.7? I think this is the same issue as #5645. I've submitted a PR for it.

**Comment by Sardtok** [comment]
This should be fixed in #5647, filter temporarily sets blend mode to REPLACE, which would disable swapping it back.
